armory/scenarios/librispeech_dev_clean_spectrogram.py

```diff
@@ -0,0 +1,171 @@
+"""
+Classifier evaluation within ARMORY
+"""
+
+import logging
+
+import numpy as np
+from tqdm import tqdm
+
+from armory.utils.config_loading import (
+    load_dataset,
+    load_model,
+    load_attack,
+)
+from armory.utils import metrics
+from armory.scenarios.base import Scenario
+
+logger = logging.getLogger(__name__)
+
+
+class LibrispeechDevCleanSpectrogram(Scenario):
+    def _evaluate(self, config: dict) -> dict:
+        """
+        Evaluate a config file for classification robustness against attack.
+        """
+        model_config = config["model"]
+        classifier, preprocessing_fn = load_model(model_config)
+
+        n_tbins = 100  # number of time bins in spectrogram input to model
+
+        task_metric = metrics.categorical_accuracy
+
+        # Train ART classifier
+        if not model_config["weights_file"]:
+            classifier.set_learning_phase(True)
+            logger.info(
+                f"Fitting model {model_config['module']}.{model_config['name']}..."
+            )
+            fit_kwargs = model_config["fit_kwargs"]
+            train_data_generator = load_dataset(
+                config["dataset"],
+                epochs=fit_kwargs["nb_epochs"],
+                split_type="train",
+                preprocessing_fn=preprocessing_fn,
+            )
+
+            for cnt, (x, y) in tqdm(enumerate(train_data_generator)):
+                # x is of shape (N,241,T), representing N spectrograms,
+                # each with 241 frequency bins and T time bins that's variable,
+                # depending on the duration of the corresponding raw audio.
+                # The model accepts a fixed size spectrogram, so x_trains need to
+                # be sampled.
+
+                x_seg = []
+                for xt in x:
+                    rand_t = np.random.randint(xt.shape[1] - n_tbins)
+                    x_seg.append(xt[:, rand_t : rand_t + n_tbins])
+                x_seg = np.array(x_seg)
+                x_seg = np.expand_dims(x_seg, -1)
+
+                classifier.fit(
+                    x_seg,
+                    y,
+                    batch_size=config["dataset"]["batch_size"],
+                    nb_epochs=1,
+                    verbose=True,
+                )
+
+                if cnt % train_data_generator.batches_per_epoch == 0:
```

Comment:
Won't this do the validation at the beginning of the epoch, after the first batch? Would we actually want
```if (cnt + 1) % train_data_generator.batches_per_epoch == 0:```